src/ol/format/geojsonformat.js

```diff
@@ -159,6 +160,16 @@ ol.format.GeoJSON.readMultiPolygonGeometry_ = function(object) {
   return new ol.geom.MultiPolygon(object.coordinates);
 };
 
+/**
+ * @param {GeoJSONGeometry} object Object.
+ * @private
+ * @return {ol.geom.Curve} Curve.
+ */
+ol.format.GeoJSON.readGeometryCurve_ = function(object) {
+  goog.asserts.assert(object.type == 'Curve',
+    'object.type should be Curve');
+  return new ol.geom.Curve(object.coordinates,object.radius);
+};
```

Comment:
From the [GeoJSON spec](https://datatracker.ietf.org/doc/draft-ietf-geojson/?include_text=1):
> 7\.  GeoJSON Types are not Extensible
>
>   Implementations MUST NOT extend the fixed set of GeoJSON types:
>   FeatureCollection, Feature, Point, LineString, MultiPoint, Polygon,
>   MultiLineString, MultiPolygon, and GeometryCollection.

We need to decide what to do when *writing* a `ol.geom.Curve` with the GeoJSON format (e.g. throw, omit it, write a LineString?).  But we should not add a reader since GeoJSON cannot have a "Curve" type geometry.